In [22]:
import import_ipynb
import preprocessing
import names_hider
import pandas as pd
import re
import os

importing Jupyter notebook from names_hider.ipynb
importing Jupyter notebook from meta_extraction.ipynb


In [2]:
PATH_TO_DATA = 'data/papaver/'

In [3]:
files = [os.path.join(PATH_TO_DATA, file) for file in os.listdir(PATH_TO_DATA)]

In [6]:
files

['data/papaver/1053.txt',
 'data/papaver/1047.txt',
 'data/papaver/289.txt',
 'data/papaver/262.txt',
 'data/papaver/276.txt',
 'data/papaver/1084.txt',
 'data/papaver/1251.txt',
 'data/papaver/909.txt',
 'data/papaver/1523.txt',
 'data/papaver/1292.txt',
 'data/papaver/1443.txt',
 'data/papaver/29.txt',
 'data/papaver/1325.txt',
 'data/papaver/869.txt',
 'data/papaver/1331.txt',
 'data/papaver/1457.txt',
 'data/papaver/15.txt',
 'data/papaver/1319.txt',
 'data/papaver/841.txt',
 'data/papaver/114.txt',
 'data/papaver/100.txt',
 'data/papaver/1494.txt',
 'data/papaver/128.txt',
 'data/papaver/1127.txt',
 'data/papaver/1641.txt',
 'data/papaver/1655.txt',
 'data/papaver/1133.txt',
 'data/papaver/316.txt',
 'data/papaver/302.txt',
 'data/papaver/303.txt',
 'data/papaver/317.txt',
 'data/papaver/1654.txt',
 'data/papaver/1132.txt',
 'data/papaver/1640.txt',
 'data/papaver/129.txt',
 'data/papaver/897.txt',
 'data/papaver/1495.txt',
 'data/papaver/101.txt',
 'data/papaver/115.txt',
 'data/

In [4]:
judices_texts = [open(file).read() for file in files]

In [5]:
len(judices_texts)

973

In [6]:
def clean(text):
    
    pattern1 = re.compile(r'(п|ч|ст)(\.|\s|\d)')
    pattern2 = re.compile(r'((У|Г)П?К|КоАП|ПДД)\sРФ')
    pattern3 = re.compile('\d')
    pattern4 = re.compile('(ДД.ММ.ГГ|дд.мм.гг)')

    text = re.sub(pattern1, '', text)
    text = re.sub(pattern2, '', text)
    text = re.sub(pattern3, '', text)
    text = re.sub(pattern4, '', text)
    text = text.replace('\xad', '')
    text = text.replace('п р и г о в о р и л', '')
    text = text.replace('\ufeff', '')
    text = text.replace('\ufeff', '')
    text = text.replace('п о с т а н о в и л', '')
    text = text.replace('П Р И Г О В О Р И Л', '')
    text = text.replace('П О С Т А Н О В И Л', '')
    
    return text

In [7]:
judices_texts = [clean(text) for text in judices_texts]

In [8]:
judices_texts[:5]

['П Р И Г О В О Р\n\n\n\n\n\n\n\n\nИМЕНЕМ РОССИЙСКОЙ ФЕДЕРАЦИИ\n\n\n\n\\intb \n\n\n\n\nг. Нягань ...\n\n\n\n\n\n\n\n\nНяганский городской суд Ханты-Мансийского автономного округа -Югры в составе председательствующего судьи Наам Е.В.,\n\n\n\n\n\n\n\n\nс участием государственного обвинителя - помощника прокурора г. Нягани ДубаниЛ.М., \n\n\n\n\n\n\n\n\nподсудимого Минаева Андрея Александровича,\n\n\n\n\n\n\n\n\nзащитника - адвоката Соротокиной И.А., представившей удостоверение Номер обезличен и ордер Номер обезличен \n\n\n\n\n\n\n\n\nпри секретаре Бородиной Н.В.,\n\n\n\n\n\n\n\n\nрассмотрев материалы уголовного дела в отношении \n\n\n\n\n\n\n\n\n\\accusedb МИНАЕВА Андрея Александровича \\accusede, родившегося ... проживающего по адресу: ..., судимого ... по    ,   ,   к  годам лишения свободы с отбыванием наказания в ИК строгого режима, освободившегося Дата обезличена по отбытию срока наказания; ... по      В,   к  годам лишения свободы условно с испытательным сроком  года, обвиняемого в 

In [9]:
trash = r'\ufeff|DIGIT|\\n|ABBR|DOCUMENT|DATE'

In [10]:
judices_texts = [re.sub(trash, ' ', text) for text in judices_texts]

In [11]:
judices_texts[:5]

['П Р И Г О В О Р\n\n\n\n\n\n\n\n\nИМЕНЕМ РОССИЙСКОЙ ФЕДЕРАЦИИ\n\n\n\n\\intb \n\n\n\n\nг. Нягань ...\n\n\n\n\n\n\n\n\nНяганский городской суд Ханты-Мансийского автономного округа -Югры в составе председательствующего судьи Наам Е.В.,\n\n\n\n\n\n\n\n\nс участием государственного обвинителя - помощника прокурора г. Нягани ДубаниЛ.М., \n\n\n\n\n\n\n\n\nподсудимого Минаева Андрея Александровича,\n\n\n\n\n\n\n\n\nзащитника - адвоката Соротокиной И.А., представившей удостоверение Номер обезличен и ордер Номер обезличен \n\n\n\n\n\n\n\n\nпри секретаре Бородиной Н.В.,\n\n\n\n\n\n\n\n\nрассмотрев материалы уголовного дела в отношении \n\n\n\n\n\n\n\n\n\\accusedb МИНАЕВА Андрея Александровича \\accusede, родившегося ... проживающего по адресу: ..., судимого ... по    ,   ,   к  годам лишения свободы с отбыванием наказания в ИК строгого режима, освободившегося Дата обезличена по отбытию срока наказания; ... по      В,   к  годам лишения свободы условно с испытательным сроком  года, обвиняемого в 

In [16]:
def get_part(text, start_point, end_point):
    beg = text.split(start_point)[1:]
    res = [elem.split(end_point)[0] for elem in beg]
    return res

In [17]:
meditations = []
fabulas = []
proves = []
witnesses = []

for text in judices_texts:
    meditations.extend(get_part(text, 'meditationb', 'meditatione'))
    fabulas.extend(get_part(text, 'fabb', 'fabe'))
    proves.extend(get_part(text, 'proofb', 'proofe'))
    witnesses.extend(get_part(text, 'witnessb', 'witnesse'))
    
    
print(len(meditations), len(fabulas), len(proves), len(witnesses))

1269 1383 828 506


In [18]:
meditations[:5]

[' Действия Минаева А.А. квалифицируются судом по     , как незаконное изготовление и хранение без цели сбыта наркотических средств в крупном размере; а также по статье    , как организация и содержание притона для потребления наркотических средств. \n\n\n\n\n\n\n\n\nПри назначении вида и размера наказания суд учитывает характер и степень общественной опасности совершенных преступлений, личность подсудимого и обстоятельства, влияющие на наказание.\n\n\n\n\n\n\n\n\nПолное признание подсудимым своей вины, признается судом обстоятельством, смягчающим наказание.\n\n\n\n\n\n\n\n\nОбстоятельством, отягчающим наказание подсудимого, является рецидив преступлений.\n\n\n\n\n\n\n\n\nВ процессе изучения личности подсудимого установлено, что он не работает, состоит у врача нарколога с диагнозом эпизодическое употребление опиатов, по месту жительства характеризуется посредственно, к административной ответственности за нарушение общественного прядка не привлекался (том л.д. , , , ).\n\n\n\n\n\n\n\n\n

In [19]:
fabulas[:5]

[' Минаев А.А. незаконно изготовил и хранил без цели сбыта наркотическое средство в крупном размере, а также организовал и содержал притон для потребления наркотических средств.\n\n\n\n\n\n\n\n\nПреступления совершены им в ... в период условного осуждения по приговору ... при следующих обстоятельствах. \n\n\n\n\n\n\n\n\nДата обезличена около  часов  минут Минаев А.А., находясь в своей квартире, расположенной по адресу: ..., без цели сбыта незаконно изготовил из семян мака и маковой соломы путем воздействия на них аммиаком и растворителем с последующим нагреванием, выпариванием и фильтрацией наркотическое средство в крупном размере - экстракт маковой соломы массой , грамма, которое без цели сбыта незаконно хранил по месту своего жительства до изъятия его сотрудниками милиции в тот же день после  часов  минут. \n\n\n\n\n\n\n\n\nКроме того, Минаев А.А. проживая по месту своей регистрации по адресу: ..., создал своими действиями необходимые условия для регулярного потребления наркотических

In [20]:
proves[:5]

[' \nВина их объективно подтверждена материалами дела: протоколами об административном правонарушении, рапортами сотрудников полиции, не доверять которым у суда не имеется оснований. Предоставленные доказательства не противоречат друг другу, в связи с чем, признаются допустимыми по делу. \\',
 ' Кроме собственных объяснений вина Саус М.А. доказана материалами дела: рапортом сотрудника полиции Прокопенко А.В., который выявил правонарушение, протоколами осмотра места происшествия, протоколом об административном правонарушении. \\',
 '  Фактические обстоятельства дела подтверждаются собранными по делу доказательствами: протоколом об административном правонарушении (л.д. ), пояснениями Лебедева А.А., рапортами должностных лиц (л.д.  - ), обращениями Астраханского отделения КПРФ (л.д.,), распоряжением администрации (л.д.), объяснениями лиц, присутствовавших на митинге (л.д.-), пояснениями свидетелей Гиллера А.М., Попова А.П., данными в ходе судебного разбирательства. \\',
 ' Вещественные до

In [21]:
witnesses[:5]

[' Как пояснил свидетель Гиллер А.М.- начальник управления координации деятельности по обеспечению общественной безопасности администрации г.Астрахани, по указанным основаниям ( в связи тем, что расстояние от ограждения Братского сада до медицинской организации составляет менее  м), по протесту прокурора было отменено аналогичное распоряжение администрации г.Астрахани, которым ранее было согласовано проведение митинга на территории Братского сада.  \\',
 ' \n\n\n\n\n\n\nПоказаниями свидетеля Т.  о том, что  апреля  года, им совместно с сотрудниками УФСКН А. , В. и Б. , с участием специалиста, на основании постановления суда и поручения следователя, был произведен обыск по адресу: \xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0..., где проживает Березовский Ю.Д. На момент начала производства обыска, в указанном домовладении находился Березовский Ю.Д. и его мать М.  Березовскому Ю.Д. предъявлено постановление суда о про

In [76]:
def remove_trash(text):
    trash = r'\n|ACCUSED|[.,;?!\'№:():-]|\\xa0|ADVOCATE|JUDGE|SECRETARY|PROSECUTOR|pleaguilty|condb|conde|\xa0|\\\\'
    no_trash_text = re.sub(trash, ' ', text)
    no_space_sequences_text = re.sub(' +', ' ', no_trash_text)
    return no_space_sequences_text

In [61]:
meditations = [names_hider.hide_all_names(text) for text in meditations]

KeyboardInterrupt: 

In [77]:
meditations = [remove_trash(text) for text in meditations]

In [78]:
meditations[:5]

[' Действия квалифицируются судом по как незаконное изготовление и хранение без цели сбыта наркотических средств в крупном размере а также по статье как организация и содержание притона для потребления наркотических средств При назначении вида и размера наказания суд учитывает характер и степень общественной опасности совершенных преступлений личность подсудимого и обстоятельства влияющие на наказание Полное признание подсудимым своей вины признается судом обстоятельством смягчающим наказание Обстоятельством отягчающим наказание подсудимого является рецидив преступлений В процессе изучения личности подсудимого установлено что он не работает состоит у врача нарколога с диагнозом эпизодическое употребление опиатов по месту жительства характеризуется посредственно к административной ответственности за нарушение общественного прядка не привлекался том л д Совершение умышленных преступлений средней тяжести в течение испытательного срока является основанием в силу для его отмены с назначение

In [27]:
fabulas = [names_hider.hide_all_names(text) for text in fabulas]

In [28]:
proves = [names_hider.hide_all_names(text) for text in proves]

In [79]:
proves = [remove_trash(text) for text in proves]

In [30]:
witnesses = [names_hider.hide_all_names(text) for text in witnesses]

In [80]:
witnesses = [remove_trash(text) for text in witnesses]

In [81]:
fabulas = [remove_trash(text) for text in fabulas]

In [82]:
fabulas[:5]

[' незаконно изготовил и хранил без цели сбыта наркотическое средство в крупном размере а также организовал и содержал притон для потребления наркотических средствПреступления совершены им в в период условного осуждения по приговору при следующих обстоятельствах Дата обезличена около часов минут находясь в своей квартире расположенной по адресу без цели сбыта незаконно изготовил из семян мака и маковой соломы путем воздействия на них аммиаком и растворителем с последующим нагреванием выпариванием и фильтрацией наркотическое средство в крупном размере экстракт маковой соломы массой грамма которое без цели сбыта незаконно хранил по месту своего жительства до изъятия его сотрудниками милиции в тот же день после часов минут Кроме того проживая по месту своей регистрации по адресу создал своими действиями необходимые условия для регулярного потребления наркотических средств несколькими лицами организовав в помещении своей квартиры притон и обеспечивал его постоянное функционирование вплоть 

In [86]:
meditations_dict = {'text': meditations, 'tag':'meditation'}

In [89]:
meditations_dict = dict.fromkeys(meditations, 'meditation')

In [93]:
df0 = pd.DataFrame(columns=['text', 'tag'])
df0['text'] = meditations
df0['tag'][:len(meditations)] = 'meditation'

In [94]:
df0.head()

text         tag
0   Действия квалифицируются судом по как незакон...  meditation
1   Действия квалифицируются по как содержание пр...  meditation
2   Действия Маркелова А А Брылева Р В Левчика И ...  meditation
3   Действия правонарушителя Саус М А квалифициро...  meditation
4   Данные признаки также нашли свое подтверждени...  meditation

In [96]:
df1 = pd.DataFrame(columns=['text', 'tag'])
df1['text'] = fabulas
df1['tag'][:len(meditations)] = 'fabula'

In [97]:
df2 = pd.DataFrame(columns=['text', 'tag'])
df2['text'] = proves
df2['tag'][:len(meditations)] = 'prove'

In [98]:
df3 = pd.DataFrame(columns=['text', 'tag'])
df3['text'] = witnesses
df3['tag'][:len(meditations)] = 'witness'

In [104]:
data = pd.concat([df0, df1, df2, df3])

In [111]:
data.count()

text    3986
tag     3872
dtype: int64

In [144]:
data.text[:1000]

1227     Совокупностью вышеприведенных доказательств с...
77       Показаниями свидетеля «под псевдонимом ФИО » ...
400      ГГ примерно в ГГ БР находясь сзади гаражей ра...
150      Показаниями свидетеля ФИО о том что ГГ они с ...
741      Таким образом общая сумма оплаты труда адвока...
274      протокол осмотра лица выступающего в качестве...
1036     Совершенное подсудимым преступление относится...
604      У суда не вызывает сомнения факт вменяемости ...
595      Актом обследования помещений территорий транс...
792      Из вышеуказанных доказательств следует что пр...
195      Начальником отделения по исполнению администр...
233      Явившаяся в суд в качестве свидетеля ФИО пред...
764      протоколом осмотра предметов от ГГ согласно к...
694      Суд приходит к выводу что обвинение с которым...
258      Хунов ВБ обвиняется в организации и содержани...
601      Берхамова ИХ незаконно содержала притон для п...
21       Соснин АВ незаконно приобрел хранил без цели ...
1059     ФИО с

In [130]:
data = data.dropna()

In [131]:
data.count()

text    3872
tag     3872
dtype: int64

In [132]:
from sklearn.utils import shuffle
from sklearn.model_selection import train_test_split

data = shuffle(data)

train, test = train_test_split(data, test_size=0.2)
train.tail()

text         tag
208   Из показаний Супрунова В М на стадии предвари...     witness
661   При разрешении вопросов касающихся преступнос...  meditation
29    Вина подсудимых также подтверждается Материал...       prove
504   Таким образом событие административного право...  meditation
652   Из протокола об административном правонарушен...      fabula

In [133]:
from sklearn.feature_extraction.text import *
from sklearn.metrics import *
from sklearn.pipeline import Pipeline
from sklearn.naive_bayes import MultinomialNB
from sklearn.linear_model import SGDClassifier, LogisticRegression
from sklearn.ensemble import RandomForestClassifier

In [134]:

vectorizer = CountVectorizer(ngram_range=(1,1))
classifier = MultinomialNB()

clf = Pipeline([
    ('vect', vectorizer),
    ('clf', classifier),
])

# учим классификатор
%time clf.fit(train.text, train.tag)

CPU times: user 1.25 s, sys: 93.6 ms, total: 1.35 s
Wall time: 1.78 s


Pipeline(memory=None,
     steps=[('vect', CountVectorizer(analyzer='word', binary=False, decode_error='strict',
        dtype=<class 'numpy.int64'>, encoding='utf-8', input='content',
        lowercase=True, max_df=1.0, max_features=None, min_df=1,
        ngram_range=(1, 1), preprocessor=None, stop_words=None,
        strip_accents=None, token_pattern='(?u)\\b\\w\\w+\\b',
        tokenizer=None, vocabulary=None)), ('clf', MultinomialNB(alpha=1.0, class_prior=None, fit_prior=True))])

In [135]:
predictions = clf.predict(test.text)

In [137]:

# макро-усреднение
print("Precision: {0:6.2f}".format(precision_score(test.tag, predictions, average='macro')))
print("Recall: {0:6.2f}".format(recall_score(test.tag, predictions, average='macro')))
print("F1-measure: {0:6.2f}".format(f1_score(test.tag, predictions, average='macro')))

print("...")
# микро-усреднение
print("Precision: {0:6.2f}".format(precision_score(test.tag, predictions, average='micro')))
print("Recall: {0:6.2f}".format(recall_score(test.tag, predictions, average='micro')))
print("F1-measure: {0:6.2f}".format(f1_score(test.tag, predictions, average='micro')))

print("...")
# accuracy
print("Accuracy: {0:6.2f}".format(accuracy_score(test.tag, predictions)))

Precision:   0.88
Recall:   0.87
F1-measure:   0.87
...
Precision:   0.88
Recall:   0.88
F1-measure:   0.88
...
Accuracy:   0.88


In [138]:
print(classification_report(test.tag, predictions))

             precision    recall  f1-score   support

     fabula       0.89      0.85      0.87       244
 meditation       0.87      0.95      0.91       251
      prove       0.90      0.82      0.86       160
    witness       0.85      0.87      0.86       120

avg / total       0.88      0.88      0.88       775



In [139]:
import pymorphy2

morph = pymorphy2.MorphAnalyzer()

In [140]:
def lemmatize(text):
    words = [morph.parse(word)[0].normal_form for word in text]
    return words

In [ ]:
data['lemmatized_text'] = data['text'].apply(lemmatize)

In [143]:
data[text][:5]

KeyError: 'Дело № -/\n\n\n\nПриговор\n\n\n\nИменем Российской Федерации\n\\intb \n\n\nселение Октябрьское \xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0 \xa0\xa0\xa0\xa0 февраля  года \n\n\n\nПригородный районный суд Республики Северная Осетия - Алания в составе:\n\n\n\nпредседательствующего судьи ФИО \n\n\n\nпри секретаре судебного заседания ФИО,\n\n\n\nс участием:\n\n\n\nгосударственных обвинителей - помощников прокурора <адрес> РСО – Алания ФИО, ФИО\n\n\n\nподсудимого \\accusedb Гойгова Р.М. \\accusede,\n\n\n\nзащитника подсудимого - адвоката Адвокатской палаты РСО - Алания Караевой Б.И., представившей удостоверение №  и ордер №  от .. года,\n\n\n\nобщественного защитника Касаевой Л.А.\n\n\n\nрассмотрев в открытом судебном заседании материалы уголовного дела в отношении Гойгова ФИО, <данные изъяты>, зарегистрированного и проживающего в <адрес>, обвиняемого в совершении преступления, предусмотренного   ,\n\n\\inte\n\nу с т а н о в и л:\n\n\n\n\\fabb Гойгов ФИО незаконно хранил без цели сбыта наркотическое средство в крупном размере при следующих обстоятельствах:\n\n\n\n.. года примерно в  часов  минут, на СП ДПС «<данные изъяты>» с. Чермен, Пригородного района, РСО - Алания, сотрудниками полиции была остановлена автомашина <данные изъяты>, черного цвета, с государственным регистрационным знаком <данные изъяты> под управлением Гойгова Р.М., и подвергнута досмотру, в ходе которого в багажном отсеке автомашины, было обнаружено и изъято наркотическое средство «экстракт маковой соломки» весом , гр. (заключению эксперта №  от .. года), которое согласно Постановлению Правительства РФ №  от .. года « Об утверждении крупного и особо крупного размеров наркотических средств и психотропных веществ, для целей статей , . и  Уголовного кодекса РФ», является крупным размером наркотического средства. Вышеуказанное наркотическое средство Гойгов Р.М. в нарушении Федерального Закона РФ №  от .. года «О наркотических средствах и психотропных веществ» без цели сбыта незаконно приобрел и хранил при себе для личного потребления.\n\n\n\nДопрошенный в судебном заседании Гойгов Р.М. виновным себя признал \\pleaguilty , от дачи подробных показаний отказался. \\fabe\n\n\n\nПомимо признания своей вины, виновность Гойгова Р.М.. в совершении преступления, предусмотренного    , полностью подтверждается совокупностью исследованных судом доказательств, а именно: \n\n\n\n\xa0\xa0\\proofb \xa0\xa0Оглашенными показаниями подозреваемого Гойгов ФИО показал, что .. года находился в г.Назрань Республики Ингушетия на автомашине <данные изъяты>, черного цвета с г<данные изъяты>, и направляясь в с.Чермен Пригородного района примерно в  часов  минут При выезде с г.Назрани не доезжая до «<данные изъяты>» примерно остановившись на обочине дороги обнаружил в кустах три бутылки с семенами кондитерского мака и одну бутылку с жидкостью коричневого цвета с характерным запахом и поняв, что там находится наркотическое средство изготовленное из семян кондитерского мака решил забрать наркотическое средство для личного употребления, в связи с чем положил обнаруженные бутылки в багажный отсек автомашины, а еще одну бутылку с растворителем и бутылку с уксусом и две миски. Сев обратно в автомашину поехал в сторону с.Чермен Пригородного района РСО – Алания через с.Майское. Подъехав к посту СП ДПС «<данные изъяты>» с.Чермен Пригородного района РСО – Алания со стороны с.Майское Пригородного района РСО – Алания был остановлен сотрудники ДПС. Остановившись для проверки документов сотрудниками полиции ему было предложено заехать во внутренний двор СП ДПС. На поСП ДПС сотрудниками полиции были приглашены двое гражданских лиц в присутствии которых ему сообщили, о том, что будет проводиться досмотр автомашины <данные изъяты>, с г/н. <данные изъяты>. До начала производства досмотра автомашины <данные изъяты>, с г/н. <данные изъяты>, ему было предложено назвать свои полные данные и добровольно выдать, все предметы, вещества и средства, незаконно хранящиеся в машине, на что он назвав свои полные данные пояснил, что ничего незаконного в машине нет. Затем, в присутствии двух участвующих лиц, сотрудники полиции стали проводить досмотр автомашины <данные изъяты>, черного цвета с г/н. <данные изъяты>, и в результате которого в багажном отсеке автомашины была обнаружена и изъята бутылка с жидкостью коричневого цвета. На вопрос сотрудника полиции, что это за жидкость и кому она принадлежит, он пояснил, что бутылке находиться наркотическое средство изготовленное из семян кондитерского мака, которую .. года примерно в  часов  минут нашел на окраине <адрес> Республики Ингушетия в районе «<данные изъяты>» и взял для личного употребления. В присутствии участвующих лиц изъятую бутылку с жидкостью коричневого цвета, упаковали в черный полиэтиленовый пакет и опечатали печатью №  «Для пакетов МВД РСО – Алания», на печати которой расписался он и двое участвующих лиц. После чего был составлен протокол досмотра автомашины, где расписались все участвующие лица и он сам. В ходе проведения досмотра автомашины <данные изъяты>, черного цвета, с г/н. <данные изъяты>, от участвующих лиц никаких заявлений, замечаний, дополнений и претензий не поступало. Никакого морального и физического давления со стороны сотрудников полиции на него не оказывалось. Претензий к сотрудникам полиции не имеет. В содеянном им преступлении раскаялся, вину признал полностью. том №, л.д - \\proofe\n\n\n\n\xa0\xa0\xa0\xa0\n\n\n\n\\witnessb Показаниями свидетеля ФИО, показал, что  июня  года, он был остановлен сотрудником ДПС и приглашен для участия в досмотре автомашине в качестве понятого. Досмотр автомашины <данные изъяты>, черного цвета, с г/н. <данные изъяты>, принадлежащий Гойгову ФИО, <адрес> проводился на территории СП ДПС <данные изъяты> <адрес> РСО – Алания. Перед началом досмотра автомашины сотрудник полиции спросил у Гойгова Р.М. имеются ли в автомашине предметы и вещества, запрещенные к свободному обороту на территории РФ, на что тот пояснил, что таковых не имеет. После этого в ходе досмотра автомашины <данные изъяты>, черного цвета с г/н. <данные изъяты> в багажном отсеке была обнаружена пластиковая бутылка с жидкостью коричневого цвета. На вопрос, что эта за жидкость, кому принадлежит и где он его приобрел, Гойгов Р.М. пояснил, что это наркотическое средство изготовленное из семян кондитерского мака, принадлежат ему, которое .. года примерно в  часов  минут он нашел на окраине г.Назрани Республики Ингушетия в районе «<данные изъяты>» и взял для личного употребления. том №, л.д. -\n\n\n\n!!!Показаниями свидетеля Дарчиев Константин Владимировидал аналогичные показания. том №, л.д.-\n\n\n\nПоказаниями свидетеля ФИО, показал, что занимает должность инспектора ИДПС  взвода роты  батальона ПДПС ГИБДД МВД по РСО–Алания, имеет специальное звание лейтенанта полиции. .. года примерно в  часов  минут заступил на суточное дежурство на СП ДПС «Кизляр» Пригородного района РСО–Алания в качестве старшего направления с.Майское Пригородного района РСО-Алания. Примерно в   минут со стороны с.Майское Пригородного района РСО – Алания к посту подъехала автомашина <данные изъяты>, черного цвета, с г/н. <данные изъяты>. После проверки документов мною было принято решение о проведении осмотра данного транспортного средства на предмет наличия наркотических средств, вследствие чего младший инспектор–кинолог ЦКС МВД по РСО–Алания, старший сержант полиции применил служебно - розыскную собаку по кличке «<данные изъяты>», по поиску и обнаружения наркотических средств и веществ, которая в ходе осмотра автомашины <данные изъяты>, черного цвета, с г/н. С  ТО  РУС обозначила сигнальным поведением на наличие в багажном отсеке с левой стороны наркотических средств и веществ. Затем водителю вышеуказанной автомашины было предложено заехать во внутрь поСП ДПС «Кизляр» для проведения досмотровых мероприятий, и в связи с чем на побыла вызвана следственна оперативная группа с Отдела МВД России по <адрес> РСО–Алания. По Прибытию следственной оперативной группы на территорию поста СП ДПС «Кизляр» были приглашены двое гражданских лиц для участия при производстве досмотровых мероприятий. Далее в присутствии двух приглашенных гражданских лиц молодому человеку было предложено сообщить свои полные данные, на что он представился Гойговым Русланом Муратовичем, ГГ года рождения, проживающим по адресу: РСО–Алания, <адрес> № . Перед началом досмотра автомашины спросили у Гойгова Р.М. имеются ли в автомашине предметы и вещества, запрещенные к свободному обороту на территории РФ, на что тот пояснил, что таковых не имеет. После этого в ходе досмотра автомашины <данные изъяты>, черного цвета с г/н. <данные изъяты> в багажном отсеке была обнаружена пластиковая бутылка с жидкостью коричневого цвета. На вопрос, что эта за жидкость, кому принадлежит и где он его приобрел, Гойгов Р.М. пояснил, что это наркотическое средство изготовленное из семян кондитерского мака, принадлежат ему, которое ГГ примерно в  часов  минут он нашел на окраине г.Назрани Республики Ингушетия в районе «<данные изъяты>» и взял для личного употребления\n\n\n\nтом №, л.д-\n\n\n\nПоказаниями свидетеля Меликов Аслан Борисовидал аналогичные показания. том №, л.д -\n\n\n\nПоказаниями свидетеля ФИО, показал, что занимает должность младшего инспектора – кинолога ЦКС МВД по РСО – Алания, имеет специальное звание старшего сержанта полиции. .. года примерно в  часов  минут заступил на дежурство на СП ДПС «Кизляр» Пригородного района РСО – Алания по направлению с.Майское. Примерно в  часов  минут со стороны с.Майское Пригородного района РСО – Алания к посту СП ДПС «Кизляр» подъехала легковая автомашина <данные изъяты>, черного цвета, с г/н. <данные изъяты>, за рулем которой сидел молодой парень, пассажиров в данной автомашине не было. Один из сотрудников ДПС остановив данную автомашину указал ему что бы он зарегистрировался, после чего водитель выйдя с машины подал документы на регистрацию. Затем было принято решение о проведении осмотра данного транспортного средства на предмет наличия наркотических средств, вследствие чего он применил служебно - розыскную собаку по кличке «Тимон», по поиску и обнаружения наркотических средств и веществ, которая в ходе осмотра автомашины <данные изъяты>, черного цвета, с г/н. <данные изъяты> обозначила сигнальным поведением ( царапанием ) на наличие в багажном отсеке с левой стороны наркотических средств и веществ. После чего сообщил об этом старшему направления потока Арчегову А.Т. Затем сотрудниками ДПС водителю вышеуказанной автомашины было предложено заехать во внутрь поСП ДПС «Кизляр» для проведения досмотровых мероприятий и в связи с чем на побыла вызвана следственна оперативная группа с ОМВД России по <адрес> РСО – Алания. том №, л.д. -\n\n\n\nПоказаниями свидетеля Тедеев Сослан Тариелович, показал, что занимает должность о/у ОУР Отдела МВД России по <адрес> РСО–А., и имеет специальное звание капитана полиции. ГГ примерно в  часов  минут по указанию оперативного дежурного по <адрес> РСО – Алания выехал в <адрес> РСО – Алания, откуда поступило сообщение о задержании автомашины <данные изъяты>, по подозрению в хранении и перевозки наркотических средств. Доехав до СП ДПС «Кизляр» и пройдя во внутрь поста увидел легковую автомашину <данные изъяты>, черного цвета, с г/н. С  ТО  РУС, рядом с которой стоял ранее незнакомый молодой человек. Сотрудники полиции пояснили, что при проверке документов служебная собака показала на наличие в данной автомашине наркотических средств. После чего было принято решение о проведении досмотра данной автомашины. С этой целью один из сотрудников поста СП ДПС «Кизляр» сев в служебную автомашину выехал в сторону <адрес> РСО – Алания, для того что бы привести двух гражданских лиц для проведения досмотра автомашины и производства личного досмотра задержанного гражданина. Спустя некоторое время на побыли приглашены двое гражданских лиц и после чего в присутствии двух участвующих лиц было предложено молодому человеку сказать свои полные данные, на что он представился Гойговым Русланом Муратовичем, ГГ года рождения, проживающим в <адрес> № . Затем Гойгову Р.М., стали разъяснять его обязанности и права при производстве досмотра принадлежащего автотранспорта. До начала досмотра Гойгову Р.М. было предложено добровольно выдать все незаконно хранящиеся предметы и вещества, находящиеся в автомашине, на что Гойгов Р.М., пояснил, что таковых нет. После чего автомашина <данные изъяты>, с г/н. <данные изъяты> был подвергнут досмотру, в результате которого в левой углу багажного отсека была обнаружена и изъята полимерная бутылка с жидкостью коричневого цвета. На вопрос, что за содержимое в полимерной бутылке, откуда и кому они принадлежат, Гойгов Р.М. пояснил, что в полимерной бутылке находиться жидкость коричневого цвета, которая является наркотическим средством изготовленной из семян кондитерского мака, принадлежат ему, которые он нашел ГГ примерно в  часов  минут на обочине дороги, расположенной на окраине <адрес> Республики Ингушетия в районе «Центра - КАМАЗ» и хранил для личного употребления. Изъятая полимерная бутылка с жидкостью коричневого цвета, была упакована в черный полиэтиленовый пакет и опечатана печатью № «Для пакетов МВД РСО–А.», на которой расписались участвующие лица и сам Гойгов Р.М. После завершения всех мероприятий были составлены акты изъятия и были получены объяснения у участвующих лиц, и у Гойгова Р.М. В ходе досмотра автомашины <данные изъяты>, черного цвета с г/н. <данные изъяты> от участвующих лиц, и самого Гойгова Р.М. претензий, заявлений, замечаний и дополнений не поступало. После завершения всех мероприятий Гойгов Р.М. был доставлен в Отдел МВД России по <адрес> РСО – Алания и передан в дежурную часть том №, л.д. -\n\n\n\nПоказаниями свидетеля Сидаков Давид Витальевидал аналогичные показания. том №, л.д. -\n\n\n\nактом изъятия вещей и предметов от ГГ автомашины <данные изъяты>, черного цвета, с г/н. <данные изъяты> принадлежащий Гойгову Р.М. на посту СП ДПС «<данные изъяты>» <адрес> РСО – Алания в присутствие двух участвующих лиц, в ходе досмотра в багажном отсеке автомашины, была обнаружена полимерная бутылка с жидкостью коричневого цвета. том № , л.д,- \\witnesse\n\n\n\n\\expertb Заключением эксперта № от «» июня  года, согласно выводам, которого, представленная на экспертизу вещество изъятая с автомашины <данные изъяты> с г/н. <данные изъяты> под управлением Гойгова Руслана Муратовича – жидкость светло-коричневого цвета, объемом , мл, находящаяся в деформированной полимерной бутылке емкостью , литра является наркотическим средством - «экстрактом маковой соломы». Масса наркотического средства – экстракта маковой соломы в пересчете на сухое вещество ( сухой остаток ), ( с учетом израсходованного на проведение экспертизы вещества ) на момент осмотра составила – , грамма. В процессе проведения исследования израсходовано по , мл исследуемых жидкостей из объектов №№ ,, и по , г вещества из объектов №№ ,,. В процессе проведения экспертизы израсходовано по , мл исследуемых жидкостей из объектов №№ ,, и по , г вещества из объектов №№ ,,. Инициатору экспертизы возвращено  мл, что в пересчете на сухое вещество составляет , г наркотического средства – «экстракт маковой соломы». том № , л.д - \\experte\n\n\n\n\\proofb Протоколом осмотра предметов от «» июня  года, согласно которого в помещении кабинета №  Отдела МВД России по Пригородному району РСО – Алания, в присутствии понятых был осмотрен черный полиэтиленовый пакет, в которой была упакована полимерная бутылка с жидкостью коричневого цвета, в которых было выявлено наркотическое средство «экстракт маковой соломы». том №, л.д -\n\n\n\nПостановлением о признании и приобщении вещественных доказательств от .. года – наркотическое средство «экстракт маковой соломы». весом – , грамм том №, л.д. \n\\proofe\n\n\n\\expertb заключением эксперта №  от .. года представленные на экспертизу следы №  и №  изъятые - с пластиковой бутылки принадлежат Гойгову Руслану Муратовичу том №, л.д. \\experte\n\n\n\n\\proofb Постановлением о признании и приобщении вещественных доказательств от .. года – бумажный конверт со следами пальцев рук. том №, л.д \n\n\n\nПротоколом №  медицинского освидетельствования, для установления факта употребления алкоголя и состояния опьянения от «» июня  года, согласно которого у Гойгова Р.М. было выявлено состояние одурманивания наркотическим веществом опийной группы. том №, л.д.  \n\n\n\n Вещественные доказательства по уголовному делу: полтора литровая полимерная бутылка с наркотическим средством «экстракт маковой соломы» хранятся в комнате хранения вещественных доказательств СО Отдела МВД России по <адрес> РСО – Алания. Бумажный конверт со следами пальцев рук обнаруженных и изъятых на полтора литровой бутылке храниться при уголовном деле № /. том № , л.д. , \\proofe\n\n\n\n<данные изъяты> \n\n\n\n\\meditationb Совокупность вышеизложенных доказательств, которые суд находит добытыми в соответствии с требованиями уголовно-процессуального закона и достоверными, с учетом чистосердечного признания Гойговым Р.М. своей вины, полностью доказывает вину подсудимого в совершении преступления, обстоятельства которого изложены в описательной части приговора. \n\n\n\nДействия подсудимого по    по признакам незаконное приобретение и хранение без цели сбыта наркотического средства в особо крупном размере квалифицированы правильно, предъявленное Гойгову Р.М. обвинение обосновано. \n\n\n\nПри назначении Гойгову Р.М. вида и размера наказания, суд исходит из требований   ,  - положений Конвенции «О защите прав человека и основных свобод» от .. года, а также - Конвенции Содружества Независимых Государств «О правах и основных свободах человека» от .. года,  ,,  и принимает во внимание характер и степень общественной опасности совершенного им преступления, которое относится к категории тяжких, обстоятельства дела, данные характеризующие его личность, в том числе смягчающие и отягчающие наказание обстоятельства, влияние назначенного наказания на исправление, а также условия жизни его семьи.\n\n\n\nПо месту жительства Гойгов Р.М. характеризуется с <данные изъяты>.\n\n\n\n<данные изъяты> \n\n\n\n<данные изъяты> \n\n\n\nГойгов Р.М. раскаялся в содеянном, полностью признал вину.Совокупность указанных обстоятельств суд признает исключительными, существенно уменьшающими степень общественной опасности преступления. \n\n\n\nОбстоятельств, отягчающих наказание подсудимого, рецидив.\n\n\n\nС учетом личности подсудимого, общественной опасности совершенного преступления и личности виновного, закрепленным в уголовном законодательстве РФ принципам гуманизма и справедливости, для достижений задаисправления осужденного и предупреждения совершения им новых преступлений в соответствии с целями наказания и принципами социальной справедливости, суд находит возможным назначить Гойгову Р.М. наказание в виде лишения свободы, но с применением положений   .\n\n\n\nВ то же время, суд не находит возможным применение в отношении подсудимого положений   . \n\n\n\nВ соответствии с «б»    Гойгову Р.М, как лицу осужденному за совершение тяжкого преступления, ранее отбывавшему лишение свободы, наказание следует отбывать в исправительной колонии строгого режима.\n\n\n\nВ связи с назначаемым наказанием, меру пресечения Гойгову Р.М в виде заключения под стражу следует оставить без изменения.\n\n\n\nСрок наказания подсудимому следует исчислять с момента его задержания. \n\n\n\nГойгову Р.М не имеет источника постоянного заработка, в связи с чем, учитывая имущественное положение осужденного, возможность получения им дохода, суд находит нецелесообразным назначение ему дополнительного наказания в виде штрафа или удержаний из заработной платы или иного дохода. \n\n\n\nВещественные доказательства: оставшееся после экспертных исследований наркотическое средство марихуана весом , гр. и два ватных тампона в соответствии с    следует уничтожить. \n\n\n\nНарушений требований Конституции Российской Федерации и «Европейской Конвенции о защите прав человека и основных свобод» при рассмотрении данного дела не установлено. \\meditatione\n\n\n\nНа основании изложенного, руководствуясь  -, , , - , суд\n\n\n\nр и г о в о р и л :\n\n\n\n\\resb \\convb Признать Гойгова <данные изъяты> виновным в совершении преступления, предусмотренного      и назначить ему наказание \\prisontermb  с применением    в виде лишения свободы сроком \\term{--} в  (два) года без штрафа и без ограничения свободы \\prisonterme, \\prisontypeb с отбыванием \\iksr в колонии строгого режима \\prisontypee.\n\n\n\nВ связи с назначаемым наказанием, меру пресечения в виде заключения под стражу, оставить без изменения, срок наказания исчислять с момента его задержания с  июня  года. \n\\conve\n\n\nВещественные доказательства: полтора литровая полимерная бутылка с наркотическим средством «экстракт маковой соломы» и два ватных– уничтожить по вступлению приговора в законную силу. \n\n\n\nПриговор может быть обжалован в апелляционном порядке в Верховный суд РСО - Алания в течение  суток со дня его провозглашения, а осужденным, содержащимся под стражей - в тот же срок со дня вручения им копии приговора. \n\n\n\nВ случае подачи апелляционной жалобы осужденный в течение  суток со дня вручении копии приговора вправе ходатайствовать о своем участии в рассмотрении уголовного дела судом апелляционной инстанции, о чем он должен указать в своей жалобе. \n\n\\rese\n\nПредседательствующий ФИО\n\n\n\nКопия верна________________'

In [ ]:

from sklearn.decomposition import TruncatedSVD

clf = Pipeline([
    ('vect', CountVectorizer(ngram_range=(1,1))),
    ('tfidf', TfidfTransformer()),
    ('reduce', TruncatedSVD(n_components=100, random_state=42)),
    ('clf', SGDClassifier())
])

%time
clf.fit(X_train, y_train)

Медитации есть в 955 текстах из 973, при этом \meditationb есть в 955 текстах
Показания свидетелей есть в 296 текстах из 973
Фабула в 946 и \\fabb в 967
Доказательства в 498 и \\proofe -- 492
Условия 167
